src/qibo/transpiler/placer.py

```diff
@@ -133,118 +117,68 @@ def __call__(self, circuit: Circuit):
                     "Gates targeting more than 2 qubits are not supported",
                 )
             if len(gate.qubits) == 2:
-                if self.middle_qubit not in gate.qubits:
+                if middle_qubit_idx not in gate.qubits:
                     new_middle = _find_connected_qubit(
                         gate.qubits,
                         circuit.queue[i + 1 :],
-                        hardware_qubits,
                         error=PlacementError,
+                        mapping=list(range(circuit.nqubits)),
                     )
-                    hardware_qubits[self.middle_qubit], hardware_qubits[new_middle] = (
-                        new_middle,
-                        self.middle_qubit,
+
+                    (
+                        circuit.wire_names[middle_qubit_idx],
+                        circuit.wire_names[new_middle],
+                    ) = (
+                        circuit.wire_names[new_middle],
+                        circuit.wire_names[middle_qubit_idx],
                     )
                     break
 
-        return dict(zip(["q" + str(i) for i in range(nqubits)], hardware_qubits))
-
 
 class Trivial(Placer):
-    """Place qubits according to the following notation:
-
-    .. math::
-        \\{\\textup{"q0"} : 0, \\textup{"q1"} : 1, ..., \\textup{"qn"} : n}.
-
-    Args:
-        connectivity (networkx.Graph, optional): chip connectivity.
-    """
+    """Place qubits according to the order of the qubit names that the user provides."""
 
     def __init__(self, connectivity: nx.Graph = None):
         self.connectivity = connectivity
 
-    def __call__(self, circuit: Circuit):
+    def __call__(self, circuit: Circuit = None):
         """Find the trivial placement for the circuit.
 
         Args:
             circuit (:class:`qibo.models.circuit.Circuit`): circuit to be transpiled.
-
-        Returns:
-            (dict): physical to logical qubit mapping.
         """
-        if self.connectivity is not None:
-            if self.connectivity.number_of_nodes() != circuit.nqubits:
-                raise_error(
-                    PlacementError,
-                    "The number of nodes of the connectivity graph must match "
-                    + "the number of qubits in the circuit",
-                )
-            trivial_layout = dict(
-                zip(
-                    ["q" + str(i) for i in list(self.connectivity.nodes())],
-                    range(circuit.nqubits),
-                )
-            )
-        else:
-            trivial_layout = dict(
-                zip(
-                    ["q" + str(i) for i in range(circuit.nqubits)],
-                    range(circuit.nqubits),
-                )
-            )
-        return trivial_layout
+        if circuit is None:
+            raise_error(ValueError, "Circuit must be provided.")
```

Comment:
This is pretty much equivalent to just drop the default value (you're just converting a `TypeError` into a `ValueError`). Moreover, the other placers have no default value.

Reply:
[f19dc6e](https://github.com/qiboteam/qibo/pull/1500/commits/f19dc6ed1ecb1e624ba6c4bbc63f72c3d3f56f20)